rust/src/api.rs

```diff
@@ -139,19 +140,19 @@ pub fn decode_invoice(invoice: String) -> Result<LightningInvoice> {
     })
 }
 
+#[derive(Clone)]
+pub enum Event {
+    Ready,
+}
+
 #[tokio::main(flavor = "current_thread")]
-pub async fn run_ldk() -> Result<()> {
+pub async fn run_ldk(stream: StreamSink<Event>) -> Result<()> {
     tracing::debug!("Starting ldk node");
-    match wallet::run_ldk().await {
-        Ok(_background_processor) => {
-            // keep connection with maker alive!
-            connection::keep_alive().await?;
-        }
-        Err(err) => {
-            tracing::error!("Error running LDK: {err}");
-        }
-    }
-    Ok(())
+    let background_processor = wallet::run_ldk().await?;
+    stream.add(Event::Ready);
+    // keep connection with maker alive!
+    connection::keep_alive().await?;
+    background_processor.join().map_err(|e| anyhow!(e))
```

Comment:
can you share the link to the docs? 😬

Reply:
I think it was more my interpretation rather than the docs spelling it out. Maybe spawning something like

```rust
loop {
    let background_processor = wallet::run_ldk().await?;
    if let Err(e) = background_processor.join() {
        tracing::error!("Background processor failed: {e:#}");
    }
}
```

would be useful.

Reply:
nice!

Reply:
I tried that, but it seems when I am moving the background_processor into a dedicated task the connection will be gone. Don't know why though.